### ISSUE TITLE
About v2 training

### ISSUE THREAD

**Issue by JiamingSuen** [lmb-freiburg/demon#20]
Hi @benjaminum, have you ever successfully trained the network with your latest v2 training code? I've been experiment with it for a few days, finding it's extremely hard to converge.
This is my training loss status:
![image](https://user-images.githubusercontent.com/8616975/30638032-b6dc36c8-9e2c-11e7-8682-5fe0869ef572.png)
(please ignore TotalLoss_1)
This is trained within about 500k iterations with default start learning rate and batch size on the first evolution. I also tried other lr and optimizer configurations, the result is similar.

1. If you do have successfully trained the network, would you like to share the hyperparameters you used?
2. I noticed that the implementation of the network is changed a little bit in `blocks.py`, could that be a reason? Why would you make this change and what is v2 means exactly..
3. What are the remaining tasks for training code, would you like to share your progress?

Thanks for this amazing work!

**Comment by benjaminum** [comment]
Hi @JiamingSuen ,

thanks for checking out our training code!

1. At the moment we use the hyperparameters as in the training code.
There is probably a lot of room for improving these parameters.
The losses will eventually converge if you train for a very long time, but it does not improve the testing performance.

2. v2 is an attempt to create a version of our network that can be trained easily with tensorflow.
It is meant as a basis for future experiments to improve the architecture.
First steps towards a better architecture are already in ```blocks.py```.
We share it because we hope it will be useful to other researchers.

3. As you probably have noticed, the training procedure is quite complex and the training losses can be difficult to understand on first glance.
One important remaining task is to provide easy to use evaluation code to better assess the network performance.



> Thanks for this amazing work!

Thank you!

**Comment by TheCrazyT** [comment]
Asking myself the same thing ... throught totalloss should go down after a while.
But it does not really look good (+160k iterations) :
https://tensorboard.dev/experiment/eVt6qfRbQkmiJ40tmRxeLA/#scalars&run=0_flow1%2Ftrainlogs
